state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -605,11 +546,14 @@ pub mod pallet {
 				Self::validate_destination_address(&destination_address, to)?;
 			let swap_origin = SwapOrigin::Vault { tx_hash };
 
+			let execute_at = frame_system::Pallet::<T>::block_number() + SWAP_DELAY_BLOCKS.into();
```

Comment:
We duplicate this in 3 places, why not just assign this inside `schedule_swap_internal` and then pass it out so it can be used in the events?